### ISSUE TITLE
Support nested macro calls in shorthand syntax

### ISSUE THREAD

**Issue by aelocson** [Jermolene/TiddlyWiki5#1541]
I often find myself wanting to embed one macro call within a parameter to another, and I have to resort to the full-blown macrocall widget, hampering readability.

A quick experiment suggests that it would be entirely possible to extend the parser to support things like this:

```
<<(macro "Some text with <<another macro call>> inside it")>>
```

… and perhaps even:

```
<<((macro "Some text with <<(another macro <<call>>)>> inside it"))>>
```

One could go deeper still, but there comes a point of diminishing returns.

The syntax of the `\define` pragma explicitly disallows the `(` character in macro names, so this proposal shouldn't break any existing WikiText.

By the way, my initial thought was to use `<<<`…`>>>`, but at the start of a line that would conflict with the syntax for quote blocks.

**Comment by aelocson** [comment]
I realise this construction would work in a slightly counterintuitive way, such that

    <<(macro "Some text with <<another macro call>> inside it")>>

would first call `macro` (with parameter `Some text with <<another macro call>> inside it`), and only once that had returned to a wikifying environment would it call `another` (with parameters `macro` and `call`).

By contrast,

    <$macrocall $name="macro" value=<<another macro call>>/>

calls `another` first.

But the result should be the same, except where certain types of JavaScript macros are involved.

**Comment by Jermolene** [comment]
Hi @aelocson I think that if we made a better macro call parser we could support simpler nesting with no special syntax required:

```
<<macro "Some text with <<another macro call>> inside it">>
```

I refactored the html element parser to operate as a sort of recursive descent parser, with the intention to refactor the others at some point. We should refactor the macro call parser in a similar way (and at the same time stop having two separate parsers for block and inline modes).

**Comment by aelocson** [comment]
Yes, if that's possible that would certainly be preferable.

**Comment by tobibeer** [comment]
This is a duplicate of #1339, perhaps close @aelocson?

**Comment by pmario** [comment]
@Jermolene or @aelocson  pls close since dupe of #1339

**Comment by pmario** [comment]
@Jermolene ... bump close?!

**Comment by flibbles** [comment]
This is **not** a duplicate of #1339.

_That_ issue is about having macros support macrocall arguments; It's a feature request. _This_ issue is about allowing text arguments to contain `>>` within them. This should already be possible without any special syntax, like @Jermolene has suggested.

Fortunately, I have a fix for this problem right here: #5451

**Comment by Jermolene** [comment]
Thanks @flibbles this can be closed now
